torch/csrc/jit/passes/onnx/unpack_quantized_weights.cpp

```diff
@@ -360,26 +406,42 @@ void unpackQuantizedWeightsHelper(
     const int64_t weight_zp = unpacked_weight.q_zero_point() + 128;
     const int64_t wt_numel = unpacked_weight.numel();
 
-    // Create caffe2::Int8GivenTensorFill node
-    std::ostringstream os;
-    for (const auto i : c10::irange(wt_numel)) {
-      os << static_cast<char>(inp_data[i] + 128);
+    if (caffe2) {
+      // Create caffe2::Int8GivenTensorFill node
+      std::ostringstream os;
+      for (const auto i : c10::irange(wt_numel)) {
+        os << static_cast<char>(inp_data[i] + 128);
+      }
+      Node* c2_weight = CreateQuantizedWeightsCaffe2(
+          os.str(), graph, wt_sizes, unpacked_weight.q_scale(), weight_zp);
+      graph->setInsertPoint(qlinear_node);
+      c2_weight->insertBefore(qlinear_node);
+      qlinear_node->insertInput(1, c2_weight->output());
+    } else {
+      std::vector<float> unpacked_weight_values;
+      unpacked_weight_values.reserve(unpacked_weight.numel());
+      auto unpacked_weight_data = reinterpret_cast<int8_t*>(unpacked_weight.data_ptr<c10::qint8>());
+      for (const auto i : c10::irange(unpacked_weight.numel())) {
+        unpacked_weight_values.push_back(static_cast<float>(unpacked_weight_data[i]));
+      }
+      std::vector<Node*> c2_weight = CreateQuantizedWeights(
+          unpacked_weight_values, graph, wt_sizes, static_cast<float>(unpacked_weight.q_scale()), weight_zp);
+      graph->setInsertPoint(qlinear_node);
+      c2_weight[0]->insertBefore(qlinear_node);
+      qlinear_node->insertInput(1, c2_weight[0]->output());
+      c2_weight[1]->insertBefore(qlinear_node);
+      qlinear_node->insertInput(2, c2_weight[1]->output());
+      c2_weight[2]->insertBefore(qlinear_node);
+      qlinear_node->insertInput(3, c2_weight[2]->output());
     }
 
-    Node* c2_weight = CreateQuantizedWeights(
-        os.str(), graph, wt_sizes, unpacked_weight.q_scale(), weight_zp);
-    graph->setInsertPoint(qlinear_node);
-    c2_weight->insertBefore(qlinear_node);
-    qlinear_node->insertInput(1, c2_weight->output());
-
     // Add bias
     at::Tensor original_bias;
     if (bias.has_value()) {
       original_bias = bias.value();
       original_bias.set_requires_grad(false);
     } else {
-      // Caffe2 ops always expect bias tensor so if not present create empty
-      // tensor.
+      // Create an empty tensor if the bias tensor is not present.
```

Comment:
I'd remove this comment entirely. Generally comments that state *what* code is doing are only needed if the code is not unusually complex.